proto/pulumi/provider.proto

```diff
@@ -456,6 +456,27 @@ message CheckRequest {
     // The type of the resource being checked. This must match the type specified by the `urn` field, and is passed so
     // that providers do not have to implement URN parsing in order to extract the type of the resource.
     string type = 7;
+
+    // Configuration for automatic resource naming behavior. This structure contains fields that control how the provider
+    // handles resource names, including proposed names and naming modes.
+    message AutonamingOptions {
+        // The proposed name for the resource being checked. This may be used by the provider as a suggestion
+        // for the final resource name, depending on the specified mode.
+        string proposedName = 1;
```

Comment:
```suggestion
        string proposed_name = 1;
```
proto files should use snake case for properties https://protobuf.dev/programming-guides/style/#message-field-names

Reply:
What about neighbour `randomSeed` and `secretNumber` a few lines above?

Reply:
they're old, and were added before we started consistently using this rule. I don't know if it's safe to rename them.

Reply:
Anyway, I updated to underscore